tools/cli/workflowCommands.go

```diff
@@ -1798,36 +1804,92 @@ func isLastEventDecisionTaskFailedWithNonDeterminism(ctx context.Context, domain
 	return false, nil
 }
 
-func getResetEventIDByType(ctx context.Context, c *cli.Context, resetType, domain, wid, rid string, frontendClient workflowserviceclient.Interface) (resetBaseRunID string, decisionFinishID int64, err error) {
+func getResetEventIDByType(
+	ctx context.Context,
+	c *cli.Context,
+	resetType, domain, wid, rid string,
+	frontendClient workflowserviceclient.Interface,
+) (resetBaseRunID string, decisionFinishID int64, err error) {
+	// default to the same runID
+	resetBaseRunID = rid
+
 	fmt.Println("resetType:", resetType)
 	switch resetType {
-	case "LastDecisionCompleted":
+	case resetTypeLastDecisionCompleted:
 		resetBaseRunID, decisionFinishID, err = getLastDecisionCompletedID(ctx, domain, wid, rid, frontendClient)
 		if err != nil {
 			return
 		}
-	case "LastContinuedAsNew":
+	case resetTypeLastContinuedAsNew:
 		resetBaseRunID, decisionFinishID, err = getLastContinueAsNewID(ctx, domain, wid, rid, frontendClient)
```

Comment:
nit: Seems like only this reset type will change baseRunID, shall we also update other util functions like `getLastDecisionCompletedID`, `getFirstDecisionCompletedID`, etc. not to return `resetBaseRunID`?

Reply:
yeah that sounds right thing to do.